webui/src/lib/components/repository/ObjectsDiff.jsx

```diff
@@ -83,16 +96,41 @@ const NoContentDiff = ({left, right, diffType}) => {
     </div>;
 }
 
-const ContentDiff = ({config, repoId, path, leftRef, rightRef, leftSize, rightSize, diffType}) => {
-    const {state} = useContext(AppContext);
+const ContentDiff = ({config, repoId, path, leftRef, rightRef, leftSize, rightSize, diffType, settings}) => {
+    const query = useMemo(() => qs({ path, presign: config.pre_sign_support_ui }),
+        [path, config.pre_sign_support_ui]);
+    const oldUrl = leftRef && buildUrl(repoId, leftRef, query);
```

Comment:
Please keep a similar API for ImageDiff and TextDiff.
I think it's better to have ImageDiff get `repoId, leftRef, rightRef, query` as params (and calculate the URLs inside),
But if you want to change the params of TextDiff, I guess it's also acceptable (but less recommended).